### PR TITLE
Enforce ScalaFix unused imports and other things during PR validation

### PR DESCRIPTION
**Pull request by lihaoyi**
Spiritual successor to #2776. This PR makes it such that no more unused imports will slip in in future, and allows them to be auto-fixed via `./mill __.fix`

Pull request: https://github.com/com-lihaoyi/mill/pull/2918

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@ jobs:
     uses: ./.github/workflows/run-mill-action.yml
     with:
       java-version: '8'
-      millargs: mill.scalalib.scalafmt.ScalafmtModule/checkFormatAll __.sources
```
Comment (lefou) [inline .github/workflows/actions.yml:98]: Can we please keep the format check. It's unrelated to unused imports.

File: build.sc [hunk 2]
```diff
@@ -317,10 +319,10 @@ trait MillPublishJavaModule extends MillJavaModule with PublishModule {
 /**
  * Some custom scala settings and test convenience
  */
-trait MillScalaModule extends ScalaModule with MillJavaModule { outer =>
+trait MillScalaModule extends ScalaModule with MillJavaModule with ScalafixModule{ outer =>
   def scalaVersion = Deps.scalaVersion
   def scalacOptions =
-    super.scalacOptions() ++ Seq("-deprecation", "-P:acyclic:force", "-feature", "-Xlint:unused")
+    super.scalacOptions() ++ Seq("-deprecation", "-P:acyclic:force", "-feature", "-Ywarn-unused:imports")
```
Comment (lefou) [inline build.sc:325]: What's the different between `-Xlint:unused` and `-Ywarn-unused`?

File: .github/workflows/actions.yml (hunk 1 above)
Comment (lihaoyi) [reply .github/workflows/actions.yml:98]: Fixed

File: build.sc (hunk 2 above)
Comment (lihaoyi) [reply build.sc]: Turns out that it's the same, I reverted to the old one

**Review by lefou — DISMISSED** [review]
Looks good to me!

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -97,6 +97,12 @@ jobs:
       java-version: '8'
       millargs: mill.scalalib.scalafmt.ScalafmtModule/checkFormatAll __.sources
 
+  scalafix-check:
+    uses: ./.github/workflows/run-mill-action.yml
+    with:
+      java-version: '8'
+      millargs: -i __.fix && git diff --exit-code
```
Comment (lefou) [inline .github/workflows/actions.yml:104]: I think scalafix has a built-in check feature

```suggestion
      millargs: -i -k __.fix --check
```

Comment (lihaoyi) [reply .github/workflows/actions.yml]: looks good

**Pull request merged**